### DISCUSSION TITLE
Create(extend) custom mjml component

### DISCUSSION THREAD

**Discussion by sweeft-vanikoakopashvili** [GrapesJS/grapesjs#5366] [Q&A]
Hi.
I am new at Grapes, and I'm trying to define a new custom component that will extend the existing one and create a new block for it to be easily drag-droppable.

```ts
  editor.DomComponents.addType('new-type', {
    isComponent: el => el.tagName === "MJ-SECTION",
    extend: 'mj-section',
    model: {
      defaults: {
        // Alos tried that *
        // ...editor.DomComponents.getType('mj-section')?.model.prototype.defaults
        attributes: {
          'padding-left': '50px',
        },
      },
    },
    view: {
      // *
      // ...editor.DomComponents.getType('mj-section')?.view.prototype,
    },
  })

```

**Block**
```ts
editor.BlockManager.add('new-type-block', {
    label: 'Custom section',
    content: {
      type: 'new-type',
      attributes: {
        'padding-left': '100px',
      },
    },
  })
```

#### But I can't drag and drop this component inside the canvas; there is an error saying
![image](https://github.com/GrapesJS/grapesjs/assets/132889487/005f0c7c-8a1a-4861-9a0c-73f46e9f03a8)

I've seen in other issues people suggesting using only blocks for defining your 'elements', but I need to attach custom behavior to sections and wrappers ... and preserve original ones as well.

**Comment by sweeft-vanikoakopashvili** [comment]
@artf Hi
I've noticed that drag-and-dropping the custom block from Block Manager doesn't work, but adding it programmatically does the job. Can you check the [sandbox](https://codesandbox.io/s/relaxed-night-5w73n3?file=/src/App.js) and test it? First, drag and drop my wrapper block from Block Manager and check the console (you will see `"Invalid target position"`), but if you select body and run 
```js
editor.getSelected().append(` <mj-wrapper data-gjs-type="custom-wrapper">
          <mj-section>
            <mj-column>
              <mj-image src="/vite.svg" /> 
            </mj-column> 
          </mj-section>
      </mj-wrapper>  `)
```
 it'll work
Maybe dropping happens outside of the mj-body tag?

**Reply by artf** [reply]
Please provide a reproducible demo

**Reply by sweeft-vanikoakopashvili** [reply]
Sorry, I missed the link.
[https://codesandbox.io/s/relaxed-night-5w73n3?file=/src/App.js](https://codesandbox.io/s/relaxed-night-5w73n3?file=/src/App.js)

**Reply by artf** [reply]
As you're defining a new `custom-wrapper` type, the `mj-body` doesn't support it, so you have to extend also `mj-body` and add the new type to `droppable` property, eg.
```js
editor.Components.addType('mj-body', {
    model: {
      defaults: {
        droppable: `[data-gjs-type="custom-wrapper"], [data-gjs-type="mj-section"], [data-gjs-type="mj-wrapper"], [data-gjs-type="mj-hero"], [data-gjs-type="mj-raw"]`,
      }
    }
  });
```

**Reply by sweeft-vanikoakopashvili** [reply]
Thank you very much.

**Comment by sweeft-vanikoakopashvili** [comment]
Here is another demo: [https://codesandbox.io/s/hungry-kapitsa-l6ms6x?file=/src/App.js](Sandbox)

I'm creating a new section component where I need to define custom attributes for all the sections, but drag and drop becomes impossible
